tests/mr/device/pool_mr_tests.cpp

```diff
@@ -126,5 +143,42 @@ TEST(PoolTest, NonAlignedPoolSize)
     rmm::logic_error);
 }
 
+TEST(PoolTest, UpstreamDoesntSupportMemInfo)
+{
+  cuda_mr cuda;
+  pool_mr mr1(&cuda);
+  pool_mr mr2(&mr1);
+  auto* ptr = mr2.allocate(1024);
+  mr2.deallocate(ptr, 1024);
+}
+
+using free_list_types =
+  ::testing::Types<mr::detail::fixed_size_free_list, mr::detail::coalescing_free_list>;
+
+TYPED_TEST_CASE(FreeListTest, free_list_types);
```

Comment:
This is bordering on white box testing. Is there no way to exercise this logic through publicly observable behavior of the pool resource?

Reply:
Yeah, it felt a little icky. Some of it (printing) might be tested when LOGGING_LEVEL is set to TRACE, but not all of it.  I was surprised that size() was not covered. Will look at it next week.

Reply:
Removed.